source/meta.tex

```diff
@@ -1354,8 +1354,14 @@
 The set of scalar types for which this condition holds is
 \impldef{which scalar types have unique object representations}.
 \begin{note}
-If a type has padding bits, the condition does not hold;
-otherwise, the condition holds true for integral types.
+If a type \tcode{T} has padding bits, the condition does not hold for \tcode{T}.
+If \tcode{T} is an integral type other than \keyword{bool}, the condition
+holds for \tcode{T}.
+The condition might not hold for \keyword{bool} because
+\keyword{true} or \keyword{false} can have multiple object representations.
+If \tcode{std::numeric_limits<T>::is_iec559} is \keyword{true}, the
+condition does not hold for \tcode{T} because there are multiple object
+representations for some values, such as zero.
```

Comment:
Do you mean plus and minus zero? Those are distinguishable.

Reply:
I believe C++ considers them to be the same value based on equality comparison. They are the same value, and positive/negative zero are two value/object representations of it.

Reply:
Well, 1/-0 yields a value different from 1/+0 in IEEE, so (regardless of equality comparison), the value +0 seems to be different from -0.

The various IEEE NaN encodings seem a better example.

Reply:
I'm not sure whether IEEE 754 floating-point formats are OK. It seems that all patterns of value representations are distinguishable in `strong_order` ([[cmp.alg] p1.3](https://eel.is/c++draft/cmp.alg#1.3)) due to the totalOrder operation.

Reply:
@frederick-vs-ja non-canonical representations are not ordered, and I believe they aren't otherwise distinguishable in any way: 
![image](https://github.com/cplusplus/draft/assets/22040976/5015595a-f0af-4725-99e2-ef55a0fbcace)

(from the IEEE-754 standard, which presumably has the same `totalOrder`)

Reply:
@jensmaurer I'm starting to think that even though `+0` and `-0` compare equal and are otherwise indistinguishable with the builtin relational operators, they are still distinct values.

I'll change the wording to talk about NaN encodings instead.

Reply:
Non-canonical representations are only present for decimal floating-point formats. I'm not sure whether decimal FP is suitable here now.

Reply:
@frederick-vs-ja then I've misunderstood what non-canonical means.

Anyhow, all major compilers yield `false` for `std::has_unique_object_representations_v<float>` on IEC 559 platforms, so consensus is that floating point numbers don't satisfy it.

The way that `totalOrder` orders different encodings of NaN could be seen as an ordering imposed on the object representation, without all those possible NaNs being distinct values.

This lack of clarity sounds like a core issue to me though, and maybe CWG should add normative wording that decides whether IEC 559 floats have unique object representations or not.